homeassistant/components/image_processing/facebox.py

```diff
@@ -75,15 +79,36 @@ def parse_faces(api_faces):
     return known_faces
 
 
-def post_image(url, image):
+def post_image(url, username, password, image):
     """Post an image to the classifier."""
     try:
         response = requests.post(
             url,
+            auth=requests.auth.HTTPBasicAuth(username, password),
             json={"base64": encode_image(image)},
             timeout=TIMEOUT
             )
-        return response
+        if response.status_code == HTTP_UNAUTHORIZED:
+            _LOGGER.error("AuthenticationError on %s", CLASSIFIER)
+        else:
+            return response
+    except requests.exceptions.ConnectionError:
+        _LOGGER.error("ConnectionError: Is %s running?", CLASSIFIER)
+
+
+def teach_file(url, username, password, name, file_path):
+    """Teach the classifier a name associated with a file."""
+    try:
+        with open(file_path, 'rb') as open_file:
+            response = requests.post(
+                url,
+                auth=requests.auth.HTTPBasicAuth(username, password),
+                data={ATTR_NAME: name, 'id': file_path},
```

Comment:
As this api is not part of home assistant, we shouldn't use constants that are used internally and across home assistant, and might change, ie `ATTR_NAME`. Using the lone constant also looks weird.

Reply:
It's ok to define and use local constants to be used with the facebox api in this module though. But then do so consistently.

Reply:
OK. My plan is to refactor these back into the `image_processing` platform